File: 3637-trionic-array-i/3637-trionic-array-i.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        p, q = False, False
        for i in range(1, len(nums)-1):
            if not p and not q and nums[i] <= nums[i - 1]:
                break
            
            if not p and not q and nums[i - 1] < nums[i] > nums[i + 1]:
                p = True
            
            if p and not q and nums[i - 1] > nums[i] < nums[i + 1]:
                q = True
            
            if p and q and nums[i] >= nums[i + 1]:
                p = False
                q = False
                break
        
        return p and q

# array
# time: O(n)
# space: O(1)

File: 3637-trionic-array-i/3637-trionic-array-i.go
func isTrionic(nums []int) bool {
    p, q := false, false
    for i := 1; i < len(nums)-1; i++ {
        if !p && !q && nums[i] <= nums[i-1] {
            break
        }

        if !p && !q && nums[i-1] < nums[i] && nums[i] > nums[i+1] {
            p = true
        }

        if p && !q && nums[i-1] > nums[i] && nums[i] < nums[i+1] {
            q = true
        }

        if p && q && nums[i] >= nums[i+1] {
            p = false
            q = false
            break
        }
    }

    if p && q {
        return true
    }
    return false
}

// array
// time: O(n)
// space: O(1)